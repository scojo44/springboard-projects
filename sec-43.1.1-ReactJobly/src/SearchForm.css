.SearchForm {
  display: flex;
  margin: .5rem auto;
  width: 60%;
}
.SearchForm input {
  flex: 1;
}
