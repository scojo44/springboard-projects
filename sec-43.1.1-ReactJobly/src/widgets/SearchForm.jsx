import React, { useState } from 'react'
import './SearchForm.css'

export default function SearchForm({fieldName, returnQuery}) {
  const EMPTY_SEARCH = {
    [fieldName]: ''
  }
  const [search, setSearch] = useState(EMPTY_SEARCH);

  return (
    <form className="SearchForm" onSubmit={handleSubmit}>
      <input type="text" id={fieldName} name={fieldName} placeholder="Enter search term" onChange={handleChange} value={search[fieldName]} />
      <button type="button">Search</button>
    </form>
  );

  function handleSubmit(e) {
    e.preventDefault();
    returnQuery(search[fieldName]? search : null);
  }

  function handleChange(e) {
    setSearch(s => ({
      ...s,
      [e.target.name]: e.target.value
    }));
  }
}
