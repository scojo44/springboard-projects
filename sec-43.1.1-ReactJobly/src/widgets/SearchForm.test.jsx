import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import SearchForm from './SearchForm'

describe('SearchForm Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <SearchForm fieldName="testfield" returnQuery={x => x} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <SearchForm fieldName="testfield" returnQuery={x => x} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
