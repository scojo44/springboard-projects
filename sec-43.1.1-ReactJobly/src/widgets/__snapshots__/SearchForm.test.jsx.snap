// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchForm Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="SearchForm"
  >
    <input
      id="testfield"
      name="testfield"
      placeholder="Enter search term"
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
